### PR TITLE
Logging data source script now supports Opensearch

### PR DESCRIPTION
**Pull request by cumcke**
- Remove "Elasticsearch" verbiage from script.
- Use ES_SERVICENAME to find the appropriate pods based on LOG_SEARCH_BACKEND to make the script compatible with Opensearch.
- Changed ELASTICSEARCH_DATASOURCE environment variable to LOGGING_DATASOURCE.

### CODE REVIEW

File: monitoring/bin/deploy_monitoring_openshift.sh [hunk 1]
```diff
@@ -52,11 +52,11 @@ else
 fi
 
 # Elasticsearch Datasource for Grafana
-ELASTICSEARCH_DATASOURCE="${ELASTICSEARCH_DATASOURCE:-false}"
-if [ "$ELASTICSEARCH_DATASOURCE" == "true" ]; then
+LOGGING_DATASOURCE="${LOGGING_DATASOURCE:-false}"
+if [ "$LOGGING_DATASOURCE" == "true" ]; then
   configureElasticsearchDatasource
```
Comment (gsmith-sas) [inline monitoring/bin/deploy_monitoring_openshift.sh:57]: Does the configureElasticsearchDatasource function still exist?  I thought it was replaced by the create_logging_datasource.sh script?

**Review by gsmith-sas — COMMENTED** [review]
I had some questions...

File: monitoring/bin/deploy_monitoring_tenant.sh [hunk 2]
```diff
@@ -172,19 +172,19 @@ kubectl apply -n $VIYA_NS -f $tenantDir/serviceMonitor-sas-pushgateway-tenant.ya
 
 # Elasticsearch Datasource for Grafana
 # Moved down to make sure Grafana pods exist
-ELASTICSEARCH_DATASOURCE="${ELASTICSEARCH_DATASOURCE:-false}"
-if [ "$ELASTICSEARCH_DATASOURCE" == "true" ]; then
+LOGGING_DATASOURCE="${LOGGING_DATASOURCE:-false}"
+if [ "$LOGGING_DATASOURCE" == "true" ] || [ ]; then
```
[the remark this answers is not in the document]
Comment (cumcke) [reply monitoring/bin/deploy_monitoring_tenant.sh:176]: That second condition was when I realized that there would need to be a lot of convoluted logic required to keep both ELASTICSEARCH_DATASOURCE and LOGGING_DATASOURCE to keep it bilingual.

Those OR statements in the next push.

File: monitoring/bin/deploy_monitoring_cluster.sh [hunk 3]
```diff
@@ -69,11 +69,13 @@ if [ "$PROM_OPERATOR_CRD_UPDATE" == "true" ]; then
   log_verbose "Updating Prometheus Operator custom resource definitions"
   crds=( alertmanagerconfigs alertmanagers prometheuses prometheusrules podmonitors servicemonitors thanosrulers probes )
   for crd in "${crds[@]}"; do
+    echo "CRD = $crd"
     crdURL="https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/$PROM_OPERATOR_CRD_VERSION/example/prometheus-operator-crd/monitoring.coreos.com_$crd.yaml"
+    echo "CRD URL = $crdURL"
     if kubectl get crd $crd.monitoring.coreos.com 1>/dev/null 2>&1; then
-      kubectl replace -f $crdURL
+      kubectl replace -f $crdURL --insecure-skip-tls-verify
     else
```
[the remark this answers is not in the document]
Comment (cumcke) [reply monitoring/bin/deploy_monitoring_cluster.sh:77]: The --insecure-skip-tls-verify was added to try and troubleshoot the issue I was running into with the GitHub certificates.  Those have been removed for the next push.

File: monitoring/bin/deploy_monitoring_openshift.sh (hunk 1 above)
Comment (cumcke) [reply monitoring/bin/deploy_monitoring_openshift.sh:57]: It does not exist - it has been correctly updated to point to the create_logging_datasource script.

**Review by gsmith-sas — APPROVED** [review]
Looks good...let's let the testers loose on it.

**Pull request merged**